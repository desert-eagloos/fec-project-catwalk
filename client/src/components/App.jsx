import React, { useState, useEffect } from 'react';
import Container from 'react-bootstrap/Container';
import Col from 'react-bootstrap/Col';
import axios from 'axios';

/* --- Import CSS files --- */
import '../css/app.css';

import 'bootstrap/dist/css/bootstrap.min.css';

import QARoot from './QA/QARoot';
import Overview from './Overview/Overview';

/* --- Import Module Components --- */
import RandR from './RandR/RandR';

function App() {
  const [product, setProduct] = useState({
    id: 18201,
    campus: 'hr-bld',
    name: 'Ernesto Sweatpants',
    slogan: 'Odit dolorem nemo id tempora qui.',
    description: 'A sapiente hic. Facilis et sit voluptatem. Ex sunt reiciendis qui ut perferendis qui soluta quod.',
    category: 'Sweatpants',
    default_price: '56.00',
    created_at: '2021-02-23T05:08:00.520Z',
    updated_at: '2021-02-23T05:08:00.520Z',
    features: [
      {
        feature: 'Cut',
        value: '"Skinny"',
      },
      {
        feature: 'Cut',
        value: '"Loose"',
      },
    ],
  });

  useEffect(() => {
    const fetchData = async () => {
      const result = await axios.get('/products/18201')
        .catch();
      setProduct(result.data);
    };

    fetchData();
  }, []);

  return (
    <Container>
      <Col />
      <Col md={10}>
        <Overview product={product} />
        <QARoot />
        <RandR id={18201} />
      </Col>
      <Col />
    </Container>
  );
}

export default App;
