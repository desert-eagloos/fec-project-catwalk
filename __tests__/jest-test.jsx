import React from 'react';
import renderer from 'react-test-renderer';
import App from '../client/src/components/App.jsx';

describe('Welcome (Snapshot)', () => {
  it('Welcome renders Hello World!!!', () => {
    const component = renderer.create(<App />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
