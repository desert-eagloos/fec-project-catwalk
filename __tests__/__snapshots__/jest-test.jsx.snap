// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome (Snapshot) Welcome renders App.jsx 1`] = `
<div
  className="main container"
>
  <div
    className="mb-4 mt-4 row"
  >
    <header>
      <h1
        className="page-branding logo page-banner"
      >
        PROJECT CATWALK
      </h1>
    </header>
  </div>
  <form
    className=""
  >
    <div
      className="form-group"
    >
      <label
        className="form-label"
        htmlFor="searchNewProductById"
      >
        Search New Product By ID
      </label>
      <input
        className="form-control"
        id="searchNewProductById"
        onChange={[Function]}
        placeholder="Enter Product ID"
        type="number"
      />
    </div>
    <button
      className="btn btn-secondary"
      disabled={false}
      onClick={[Function]}
      type="submit"
    >
      Submit
    </button>
  </form>
  <div
    className="container"
  >
    <div
      className="mb-4 row"
    />
    <div
      className="mb-4 row"
    >
      <div>
        <div
          className="container-fluid"
        >
          <div
            className="row"
          >
            <div
              className="col"
            >
              <h3
                id="title"
              >
                Questions and Answers
              </h3>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="mb-4 row"
    >
      <div
        className="container"
      >
        <div
          className="row"
        >
          <h3
            className="text-uppercase"
          >
            Ratings and Reviews
          </h3>
        </div>
        <div
          className="row"
        >
          <div
            className="col-4 ratings col"
          >
            <div>
              Loading Ratings...
            </div>
          </div>
          <div
            className="col-8 reviews col"
          >
            Reviews are loading...
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
